apps/gallery/js/pick.js

```diff
@@ -238,9 +240,14 @@ var Pick = (function() {
       pickType = null;   // Return unchanged or convert to JPEG
     }
 
-    if (pickType && pickType !== 'image/jpeg' && pickType !== 'image/png') {
+    if (pickType && pickType !== 'image/jpeg' &&
+        pickType !== 'image/jpeg+exif' && pickType !== 'image/png') {
       pickType = null;   // Return unchanged or convert to JPEG
     }
+    else if (!pickType && pickedFileInfo.type == 'image/jpeg') {
```

Comment:
It is deliberate. We want to set the picktype to Exif if we requested nothing and picked a JPEG. It is really the else case.